### ISSUE TITLE
[1.4.0rc*] PY3: COMM autodetection is broken

### ISSUE THREAD

**Issue by foosel** [foosel/OctoPrint#3385]
**COMM autodetection is broken**
```
Traceback (most recent call last):
  File "/home/bzed/tools/OctoPrint/src/octoprint/util/comm.py", line 2694, in _openSerial
    serial_obj = factory(self, self._port, self._baudrate, settings().getFloat(["serial", "timeout", "connection"]))
  File "/home/bzed/tools/OctoPrint/src/octoprint/util/comm.py", line 2647, in default
    port = self._detect_port()
  File "/home/bzed/tools/OctoPrint/src/octoprint/util/comm.py", line 2627, in _detect_port
    programmer.connect(p)
  File "/home/bzed/tools/OctoPrint/src/octoprint/util/avr_isp/stk500v2.py", line 36, in connect
    self.sendMessage([1])
  File "/home/bzed/tools/OctoPrint/src/octoprint/util/avr_isp/stk500v2.py", line 102, in sendMessage
    checksum ^= ord(c)
TypeError: ord() expected string of length 1, but int found
```

struct.pack returns bytes in Python3, looping over bytes gives integers there.

I've tried
```
--- a/src/octoprint/util/avr_isp/stk500v2.py
+++ b/src/octoprint/util/avr_isp/stk500v2.py
@@ -98,7 +98,7 @@ class Stk500v2(ispBase.IspBase):
                for c in data:
                        message += struct.pack(">B", c)
                checksum = 0
-               for c in message:
+               for c in message.decode():
                        checksum ^= ord(c)
                message += struct.pack(">B", checksum)
                try:
```
But then the connections just timed out. There must be another issue hidden somewhere.

_Originally posted by @bzed in https://github.com/foosel/OctoPrint/issues/3382#issuecomment-564282370_

**Comment by foosel** [comment]
Fixed in b06c222

**Comment by foosel** [comment]
1.4.0rc3 has just been released
